docs/dev/pagination-v2.md

```diff
@@ -0,0 +1,173 @@
+# Pagination in v2 Engine
+
+<!-- Why is pagination necessary? -->
+Pagination allows a SQL plugin client to retrieve arbitrarily large
+results sets one subset at a time.
+
+A cursor is a SQL abstraction for pagination. A client can open a cursor, retrieve a subset of data given a cursor and close a cursor.
+<!-- What applications will use pagination? -->
+
+Currently, SQL plugin does not provide SQL cursor syntax. However, the SQL REST endpoint can return result a page at a time.
```

Comment:
```suggestion
Currently, SQL plugin does not provide SQL cursor syntax. However, the SQL REST endpoint can return results a single page at a time.
```

Reply:
Cursor syntax is just one part of the scroll API and only one part of 'pagination' feature.  We should be specific what the SQL plugin currently provides (in legacy - link please) vs what the SQL plugin should provide (new work) vs what the OpenSearch Scroll API provides in their REST API (link please)